### DISCUSSION TITLE
How to integrate with react/angular/vue/stentil app?

### DISCUSSION THREAD

**Discussion by Eliot00** [DioxusLabs/dioxus#251] [Q&A]
If I have some component write by other frontend library, can I use them in dioxus desktop?

**Comment by t1m0t** [comment]
Could you be more specific and provide an example on what you'r trying to do?

**Reply by Eliot00** [reply]
I want to make a markdown editor, there is a [ReactJS component](https://github.com/uiwjs/react-md-editor). Is there anything I can do to get it embedded in my dioxus desktop app?

```rust
fn app(cx: Scope) -> Element {

    cx.render(rsx!(
        div { include_react { MDEditor } }
    ))
}
```

**Reply by jkelleyrtp** [reply]
You can render React directly into an Element if you get it by ID.

Something along the lines of:
- spawn a future only when the component is mounted
- query the page by ID to get the element to render into
- call React.render using the element returned above
- on each render, update your React app's internal state

There's no straightforward way of sending events from React into Dioxus (yet), so that's logic you'll need to write yourself.
